setup.py

```diff
@@ -129,9 +130,14 @@ def run(self):
         if use_system():
             return
 
-        build_temp = os.path.abspath(self.build_temp)
+        # use Python's build environment variables
+        build_env = {key: val for key, val in get_config_vars().items() if key in
+                     ("LDFLAGS", "CFLAGS", "CC", "CCSHARED", "LDSHARED")
+                     and key not in os.environ}
+        os.environ.update(build_env)
```

Comment:
I'm confused why this would be necessary. setuptools/distutils use the build flags of the compiled Python by default so whatever was used during compilation *should* be passed to the compiler/linker when building an extension. It seems you're not seeing that behavior? Am I mistaken in how setuptools works?

Reply:
setuptools/distutils do use these build flags for compiling the _extension_ (_sodium.so) but setuptools/distutils are not the ones building the _external library_ (libsodium.so). pynacl is compiling libsodium by calling configure+make directly in this function and it is not using setuptools/distutils for that compilation. When we call configure+make directly (`subprocess.check_call` a few lines down) the build environment is simply inherited from the user environment and it is not using the setuptools/distutils environments. This means _sodium.so, which uses libsodium.so, may be built with different flags and thus fail to link (`-m64` in the case of #502 and #467).

Reply:
Ah, yes, of course. My mistake!